app/domain/authentication/authn_k8s/validate_pod_request.rb

```diff
@@ -63,20 +63,32 @@ def validate_pod_exists
       end
 
       def validate_resource_restrictions
-        @validate_resource_restrictions.(
-          host_id: k8s_host.conjur_host_id,
-          host_annotations: host.annotations,
+        @validate_resource_restrictions.call(
+          authenticator_name: K8S_AUTHENTICATOR_NAME,
           service_id: service_id,
           account: k8s_host.account,
-          spiffe_id: spiffe_id
+          role_name: k8s_host.k8s_host_name,
+          constraints: Restrictions::CONSTRAINTS,
+          authentication_request: authentication_request
+        )
+      end
+
+      def authentication_request
+        @authentication_request_class.new(
+          namespace: pod_namespace,
+          validate_k8s_resource: ValidateK8sResource.new(
```

Comment:
`ValidateK8sResource` should be a dependency

Reply:
done